ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -362,23 +362,29 @@ def get_tables_name_and_type(self) -> Optional[Iterable[Tuple[str, str]]]:
     def get_view_definition(
         self, table_type: str, table_name: str, schema_name: str, inspector: Inspector
     ) -> Optional[str]:
-        if table_type in (TableType.View, TableType.MaterializedView):
-            try:
-                view_definition = inspector.get_view_definition(table_name, schema_name)
-                view_definition = (
-                    "" if view_definition is None else str(view_definition)
+        """
+        Get the DDL statement or View Definition for a table
+        """
+        try:
+            if table_type in (TableType.View, TableType.MaterializedView):
+                schema_definition = inspector.get_view_definition(
+                    table_name, schema_name
+                )
+            else:
+                schema_definition = inspector.get_table_ddl(
+                    self.connection, table_name, schema_name
                 )
-                return view_definition
+            schema_definition = (
+                "" if schema_definition is None else str(schema_definition)
```

Comment:
do we want to return an empty string? If `schema_definition` is None, let's just return the `None` and work later on with optionals no?

Empty strings can easily become error prone

Reply:
Changed.  Returns `None` or else the `schema_definition` now